shell/platform/android/io/flutter/embedding/engine/systemchannels/KeyEventChannel.java

```diff
@@ -44,11 +45,63 @@ public void keyDown(@NonNull KeyEvent keyEvent) {
     channel.send(message);
   }
 
-  private void encodeKeyEvent(@NonNull KeyEvent event, @NonNull Map<String, Object> message) {
-    message.put("flags", event.getFlags());
-    message.put("codePoint", event.getUnicodeChar());
-    message.put("keyCode", event.getKeyCode());
-    message.put("scanCode", event.getScanCode());
-    message.put("metaState", event.getMetaState());
+  private void encodeKeyEvent(@NonNull FlutterKeyEvent event, @NonNull Map<String, Object> message) {
+    message.put("flags", event.flags);
+    message.put("codePoint", event.codePoint);
+    message.put("keyCode", event.keyCode);
+    message.put("scanCode", event.scanCode);
+    message.put("metaState", event.metaState);
+    if (event.complexCharacter != null) {
+      message.put("character", event.complexCharacter.toString());
+    }
+  }
+
+  /**
+   * Key event as defined by Flutter.
+   */
+  public static class FlutterKeyEvent {
+    public final int flags;
+    public final int codePoint;
+    public final int keyCode;
+    @Nullable
+    public final Character complexCharacter;
+    public final int scanCode;
+    public final int metaState;
+
+    public FlutterKeyEvent(
+        @NonNull KeyEvent androidKeyEvent
+    ) {
+      this(androidKeyEvent, null);
+    }
+
+    public FlutterKeyEvent(
+        @NonNull KeyEvent androidKeyEvent,
+        @Nullable Character complexCharacter
+    ) {
+      this(
+          androidKeyEvent.getFlags(),
+          androidKeyEvent.getUnicodeChar(),
```

Comment:
I also need the "plainCodePoint" that was in my other PR too.  Pretty simple, but it needs to be the result of `androidKeyEvent.getUnicodeCharacter(0x0)` (i.e. same as codePoint, but called with a zeroed-out modifier mask).